pkg/master/pod_cache.go

```diff
@@ -221,6 +227,10 @@ func (p *PodCache) GarbageCollectPodStatus() {
 // calling again, or risk having new info getting clobbered by delayed
 // old info.
 func (p *PodCache) UpdateAllContainers() {
+	// TODO: this is silly, we should pro-actively update the pod status when
+	// the API server makes changes.
+	p.clearNodeStatus()
```

Comment:
Doesn't this cause the pod's phase to become unknown? Is that sufficient to cause it to be replaced by replication controller?

Why isn't the node status updated when the node becomes unresponsive?